database/src/main/postgres/flows/V1.7.5__get_flow_checkpoints.sql

```diff
@@ -0,0 +1,127 @@
+--                                that will be retrieved
+--      i_limit                 - (optional) maximum number of checkpoints retrieved
+--                                if zero or lower specified, all data will be returned, i.e. no limit will be applied
+--      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
+--
+--      Note: checkpoint name uniqueness is not enforced by the data model, so there can be multiple different
+--          checkpoints with the same name in the DB, i.e. multiple checkpoints can be retrieved even when
+--          specifying `i_checkpoint_name` parameter
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_checkpoint       - id of retrieved checkpoint
+--      checkpoint_name     - name of retrieved checkpoint
+--      measure_name        - measure name associated with a given checkpoint
+--      measure_columns     - measure columns associated with a given checkpoint
+--      measurement_value   - measurement details associated with a given checkpoint
+--      checkpoint_time     - time
+--
+-- Status codes:
+--      11                  - Flow created
+--      16                  - No checkpoints found
+--      41                  - Partitioning not found
+--
+--
+-------------------------------------------------------------------------------
+
+DECLARE
+    _fk_partitioning BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning_of_flow);
+
+    IF _fk_partitioning IS NULL THEN
+        status := 41;
+        status_text := 'The partitioning does not exist.';
+        RETURN NEXT;
+        RETURN;
+    END IF;
+
+    RETURN QUERY
+    SELECT 11 AS status, 'OK' AS status_text,
+           CP.id_checkpoint, CP.checkpoint_name,
+           MD.measure_name, MD.measured_columns,
+           M.measurement_value,
+           CP.process_start_time AS checkpoint_start_time, CP.process_end_time AS checkpoint_end_time
+    FROM runs.checkpoints AS CP
+    JOIN runs.measurements AS M
+      ON CP.id_checkpoint = M.fk_checkpoint
+    JOIN runs.measure_definitions AS MD
+      ON M.fk_measure_definition = MD.id_measure_definition
+    WHERE EXISTS (
+            -- Identification of a flow and all partitionings related to it.
+            SELECT 1
+            FROM flows.flows AS F
+            JOIN flows.partitioning_to_flow AS PF
+              ON F.id_flow = PF.fk_flow
+            WHERE F.fk_primary_partitioning = _fk_partitioning
+              AND CP.fk_partitioning = PF.fk_partitioning
+          )
+      AND CASE
+            WHEN i_checkpoint_name IS NOT NULL THEN CP.checkpoint_name = i_checkpoint_name
+            ELSE TRUE
+          END
+    ORDER BY CP.process_start_time,
+             CP.id_checkpoint
+    LIMIT CASE
+            WHEN i_limit <= 0 THEN NULL -- NULL means no limit will be applied, same as LIMIT ALL
+            ELSE i_limit
+          END;
```

Comment:
```suggestion
    LIMIT nullif(i_limit, 0)
```
Or you can make it a separate variable to make the query look shorter and easier to understand.

Reply:
If you want the negative then I would create a `_real_limit` local variable filled before the actual query based on the value of `i_limit`.